lumicks/pylake/kymotracker/detail/gaussian_mle.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+from functools import partial
+from scipy.optimize import minimize
+
+
+def poisson_log_likelihood(params, expectation_fcn, photon_count):
+    """Calculates the log likelihood of Poisson distributed values.
+
+    Parameters
+    ----------
+    params : array-like
+        Model parameters to be estimated
+    x : array-like
+        Position data at which the function is to be evaluated.
+    photon_count : array-like
+        Measured photon counts at each position to be fitted.
+    """
+    expectation = expectation_fcn(*params)
+    log_likelihood = photon_count * np.log(expectation) - expectation
+    return np.sum(-log_likelihood)
+
+
+def poisson_log_likelihood_jacobian(params, expectation_fcn, derivatives_fcn, photon_count):
+    """Evaluate the derivatives of the likelihood function w.r.t. each parameter."""
+    derivatives = derivatives_fcn(*params)
+    count_over_expectation = photon_count / expectation_fcn(*params)
+    return [-np.sum(d * ((count_over_expectation) - 1)) for d in derivatives]
+
+
+def _run_mle_optimization(initial_guess, expectation_fcn, derivatives_fcn, photon_count, bounds):
```

Comment:
I always find `run` to be and odd verb to be on a function name, I always take it to be assumed. How about `_mle_optimize`?